src/coreclr/nativeaot/Runtime/eventtrace.cpp

```diff
@@ -302,8 +325,47 @@ void EtwCallback(
         GCHeapUtilities::GetGCHeap()->DiagTraceGCSegments();
     }
 }
+
 #endif // FEATURE_ETW
 
+/**************************************************************************************/
+/* Called when ETW is turned OFF on an existing process .Will be used by the controller for end rundown*/
+/**************************************************************************************/
+void ETW::EnumerationLog::EndRundown()
+{
+    if (IsRuntimeRundownProviderEnabled(TRACE_LEVEL_INFORMATION, CLR_RUNDOWNLOADER_KEYWORD))
+    {
+        HANDLE pModule = PalGetModuleHandleFromPointer((void*)&EndRundown);
+
+        const TCHAR * wszModuleFileName = NULL;
+        const TCHAR * wszModuleILFileName = _T("");
+
+        PalGetModuleFileName(&wszModuleFileName, pModule);
+
+        GUID nativeGuid;
+        uint32_t dwAge;
+        TCHAR wszPath[1024];
+        PalGetPDBInfo(pModule, &nativeGuid, &dwAge, wszPath, ARRAY_SIZE(wszPath));
+
+        GUID zeroGuid = { 0 };
+        FireEtwModuleDCEnd_V2(
```

Comment:
Is this information only coming through the `ModuleDCEnd` event? If so, I think it should also be in a `ModuleLoad` event that gets fired on start. If a tool enables collection at before startup / for the lifetime of a process, it shouldn't have to trigger a rundown to get additional information (since it shouldn't have 'missed' anything).

Reply:
Maybe at the end of `InitializeRuntime` - after `InitializeModules`, before we call the user's managed code?